airflow/contrib/operators/gcp_container_operator.py

```diff
@@ -0,0 +1,127 @@
+        :param name: The name of the resource to delete, in this case cluster name
+        :type name: str
+        :param location: The name of the Google Compute Engine zone in which the cluster
+        resides.
+        :type location: str
+        :param gcp_conn_id: The connection ID to use connecting to Google Cloud Platform.
+        :type gcp_conn_id: str
+        :param api_version: The api version to use
+        :type api_version: str
+        """
+        super(GKEClusterDeleteOperator, self).__init__(*args, **kwargs)
+
+        self.project_id = project_id
+        self.gcp_conn_id = gcp_conn_id
+        self.location = location
+        self.api_version = api_version
+        self.cluster_name = name
+
+    def _check_input(self):
+        if not self.project_id:
+            self.log.error('Please specify the project_id of cluster to use.')
+            raise AirflowException('project_id parameter missing, unable to trigger job')
+        if not self.cluster_name:
+            self.log.error('Please specify the cluster\'s name to delete.')
+            raise AirflowException('cluster_name parameter missing, unable to start job')
+        if not self.location:
+            self.log.error('Please specify the cluster location id to use.')
+            raise AirflowException('location parameter missing, unable to trigger job')
+
+    def execute(self, context):
+        self._check_input()
+        hook = GKEClusterHook(self.project_id, self.location)
+        delete_result = hook.delete_cluster(cluster_name=self.cluster_name)
+        return delete_result
+
+
+class GKEClusterCreateOperator(BaseOperator):
+    template_fields = ['project_id', 'gcp_conn_id', 'location', 'api_version', 'body']
+
+    @apply_defaults
+    def __init__(self,
+                 project_id,
+                 location,
+                 gcp_conn_id='google_cloud_default',
+                 api_version='v2',
+                 body={},
+                 *args,
+                 **kwargs):
+        """
+        :param project_id: The Google Developers Console [project ID or project number]
+        :type project_id: str
+        :param location: The name of the Google Compute Engine zone in which the cluster
+        resides.
+        :type location: str
+        :param body: The Cluster definition to create, can be protobuf or python dict, if
+        dict it must match protobuf message Cluster
+        :type body: dict or google.cloud.container_v1.types.Cluster
+        :param gcp_conn_id: The connection ID to use connecting to Google Cloud Platform.
+        :type gcp_conn_id: str
+        :param api_version: The api version to use
+        :type api_version: str
+        """
+        super(GKEClusterCreateOperator, self).__init__(*args, **kwargs)
+
+        self.project_id = project_id
+        self.gcp_conn_id = gcp_conn_id
+        self.location = location
+        self.api_version = api_version
+        self.body = body
+
+    def _check_input(self):
```

Comment:
_check_input can be pushed to the GkeClusterHook or an "internal" function in the operator module.